api/src/test/java/org/apache/iceberg/TestPartitionPaths.java

```diff
@@ -42,8 +42,8 @@ public void testPartitionPath() {
     Transform<Integer, Integer> bucket = Transforms.bucket(10);
 
     Literal<Long> ts =
-        Literal.of("2017-12-01T10:12:55.038194").to(Types.TimestampType.withoutZone());
-    Object tsHour = hour.bind(Types.TimestampType.withoutZone()).apply(ts.value());
+        Literal.of("2017-12-01T10:12:55.038194").to(Types.TimestampType.microsWithoutZone());
+    Object tsHour = hour.bind(Types.TimestampType.microsWithoutZone()).apply(ts.value());
```

Comment:
Can you add a test for nanos? I think we also need identity tests.

Reply:
Will do.

Reply:
Done.